.github/workflows/publish-packages.yml

```diff
@@ -323,5 +307,136 @@ jobs:
 
       - name: Uninstall Hazelcast from homebrew
         run: |
-          source ./common.sh          
+          source ./common.sh
           brew uninstall ${{ env.HZ_DISTRIBUTION}}@$BREW_PACKAGE_VERSION
+
+  buildVagrant:
+    runs-on: macos-12
+    needs: [ prepare, rpm, deb ]
+    env:
+      VAGRANT_CWD: ${{ github.workspace }}/hazelcast-packaging/vagrant
+      HZ_VERSION: ${{ needs.prepare.outputs.hz_version }}
+      PACKAGE_VERSION: ${{ needs.prepare.outputs.package_version }}
+
+    defaults:
+      run:
+        working-directory: ./hazelcast-packaging
+
+    steps:
+      - name: Checkout hazelcast-packaging repo
+        uses: actions/checkout@v2.3.2
+        with:
+          path: 'hazelcast-packaging'
+
+      - name: Cache Vagrant boxes
+        uses: actions/cache@v2
+        with:
+          path: ~/.vagrant.d/boxes
+          key: ${{ runner.os }}-vagrant-${{ hashFiles('Vagrantfile') }}
+          restore-keys: |
+            ${{ runner.os }}-vagrant-
+
+      - name: Show Vagrant version
+        run: vagrant --version
+
+      - name: Run Vagrant up to cache machines
+        run: |
+          source ./common.sh
+          vagrant up
+
+      - name: Run Vagrant halt
+        run: vagrant halt
+
+  testOnVagrant:
+    runs-on: macos-12
+    needs: [prepare, buildVagrant]
+
+    defaults:
+      run:
+        working-directory: ./hazelcast-packaging
+
+    continue-on-error: true
+
+    strategy:
+      fail-fast: false
+      matrix:
+        hz_distribution: [ 'hazelcast', 'hazelcast-enterprise' ]
+        linux_distribution: [ 'fedora', 'ubuntu' ]
+    env:
+      HZ_DISTRIBUTION: ${{ matrix.hz_distribution }}
+      LINUX_DISTRIBUTION: ${{ matrix.linux_distribution }}
+      VAGRANT_CWD: ${{ github.workspace }}/hazelcast-packaging/vagrant
+      HZ_VERSION: ${{ needs.prepare.outputs.hz_version }}
+      PACKAGE_VERSION: ${{ needs.prepare.outputs.package_version }}
+
+    steps:
+      - name: Checkout hazelcast-packaging repo
+        uses: actions/checkout@v2.3.2
+        with:
+          path: 'hazelcast-packaging'
+
+      - name: Cache Vagrant boxes
+        uses: actions/cache@v2
+        with:
+          path: ~/.vagrant.d/boxes
+          key: ${{ runner.os }}-vagrant-${{ hashFiles('Vagrantfile') }}
+          restore-keys: |
+            ${{ runner.os }}-vagrant-
+
+      - name: Run Vagrant machine ${{ env.LINUX_DISTRIBUTION}}
+        run: |
+          source ./common.sh
+          vagrant up ${{ env.LINUX_DISTRIBUTION}}
+
+      - name: Run Hazelcast ${{ matrix.hz_distribution }} on ${{ matrix.linux_distribution }}
+        run: |
+          vagrant ssh ${{ env.LINUX_DISTRIBUTION}} -c \
+            'HAZELCAST_CONFIG="/home/vagrant/hz-config.yaml" hz-start > ~/hz.log 2>&1 &'
+
+      - name: Wait for cluster to be up
+        run: |
+          vagrant ssh ${{ env.LINUX_DISTRIBUTION}} -c '~/check-hazelcast-health.sh'
+
+      - name: check health via script
```

Comment:
Could we check systemd functionality?

See description of the following PRs:
- https://github.com/hazelcast/hazelcast-packaging/pull/120
- https://github.com/hazelcast/hazelcast-packaging/pull/121

Reply:
b70824b070b37746ca9a691e0cae80b9550c0921

Reply:
Could you also check other operations (start, stop, restart) also if after the removal of the package `systemctl status hazelcast` reports that service is not known